### ISSUE TITLE
SetBackgroundColour/SetForegroundColour reset Button style

### ISSUE THREAD

**Issue by Froyok** [wxWidgets/Phoenix#815]
**Operating system**: Windows 7 x64 (SP2)
**wxPython version**: 4.0.1
**Stock or custom build**: Stock (installed via pip)
**Python version**: Python 3.6.4 (v3.6.4:d48eceb, Dec 19 2017, 06:04:45)
**Stock or custom build**: Stock

**Description of the problem** :
When setting a custom foreground or background color on a button, the specified text layout for the label is ignored and will default back to centered. If no colors are specified, everything works fine. See example below.
I observed the issue via the wxGlade preview, but is also happen even when running the code as standalone.

**Minimal app to reproduce** :

```python
#!/usr/bin/env python
# -*- coding: UTF-8 -*-
#
# generated by wxGlade 0.8.0 on Sun Apr 15 16:21:17 2018
#

import wx

class MyFrame(wx.Frame):
    def __init__(self, *args, **kwds):
        kwds["style"] = kwds.get("style", 0) | wx.DEFAULT_FRAME_STYLE
        wx.Frame.__init__(self, *args, **kwds)
        self.SetSize((400, 300))
        self.button_1 = wx.Button(self, wx.ID_ANY, "button_1\nMultiline", style=wx.BU_LEFT | wx.BU_TOP)
        self.button_2 = wx.Button(self, wx.ID_ANY, "button_1\nMultiline", style=wx.BU_LEFT | wx.BU_TOP)
        self.button_3 = wx.Button(self, wx.ID_ANY, "button_1\nMultiline", style=wx.BU_LEFT | wx.BU_TOP)

        self.__set_properties()
        self.__do_layout()

    def __set_properties(self):
        self.SetTitle("frame")
        self.button_1.SetMinSize((170, 60))
        self.button_2.SetMinSize((170, 60))
        self.button_2.SetBackgroundColour(wx.Colour(255, 0, 0))
        self.button_3.SetMinSize((170, 60))

    def __do_layout(self):
        sizer_1 = wx.BoxSizer(wx.VERTICAL)
        sizer_1.Add(self.button_1, 0, 0, 0)
        sizer_1.Add(self.button_2, 0, 0, 0)
        sizer_1.Add(self.button_3, 0, 0, 0)
        self.SetSizer(sizer_1)
        self.Layout()

class MyApp(wx.App):
    def OnInit(self):
        self.frame = MyFrame(None, wx.ID_ANY, "")
        self.SetTopWindow(self.frame)
        self.frame.Show()
        return True

if __name__ == "__main__":
    app = MyApp(0)
    app.MainLoop()

```

**Result**:
The code above produce the following for me :
![demo](https://user-images.githubusercontent.com/6290387/38779540-24cdb5ea-40ca-11e8-869d-e93e72992621.png)

**Comment by RobinD42** [comment]
This is a wxWidgets issue. IIRC when you change the colors of a button it switches to owner-drawn mode and is no longer drawn by the native control. wxWidgets can probably be modified to correct this difference, but you'll need to submit a ticket to https://trac.wxwidgets.org/

**Comment by Froyok** [comment]
I see, I will try to report it there then (I'm not a really good C/C++ user unfortunately).

**Comment by swt2c** [comment]
Looks like VZ fixed it already:
https://github.com/wxWidgets/wxWidgets/commit/a6d5ddf96ce526f95b153372de7ecd8fb1da8de1

**Comment by RobinD42** [comment]
I'm updating the submodule link now...

**Comment by RobinD42** [comment]
Closed by #847
